deltacat/compute/compactor/compaction_session.py

```diff
@@ -62,57 +56,40 @@ def compact_partition(
-        delete_prev_primary_key_index: bool = False,
         pg_config: Optional[PlacementGroupConfig] = None,
         schema_on_read: Optional[pa.schema] = None,  # TODO (ricmiyam): Remove this and retrieve schema from storage API
         rebase_source_partition_locator: Optional[PartitionLocator] = None,
         rebase_source_partition_high_watermark: Optional[int] = None,
         deltacat_storage=unimplemented_deltacat_storage) -> Optional[str]:
-
     logger.info(f"Starting compaction session for: {source_partition_locator}")
     partition = None
-    compaction_rounds_executed = 0
-    has_next_compaction_round = True
     new_rcf_s3_url = None
-    while has_next_compaction_round:
-        has_next_compaction_round, new_partition, new_rci, new_rcf_s3_url = \
-            _execute_compaction_round(
-                source_partition_locator,
-                destination_partition_locator,
-                primary_keys,
-                compaction_artifact_s3_bucket,
-                last_stream_position_to_compact,
-                hash_bucket_count,
-                sort_keys,
-                records_per_primary_key_index_file,
-                records_per_compacted_file,
-                input_deltas_stats,
-                min_pk_index_pa_bytes,
-                min_hash_bucket_chunk_size,
-                compacted_file_content_type,
-                delete_prev_primary_key_index,
-                pg_config,
-                schema_on_read,
-                rebase_source_partition_locator,
-                rebase_source_partition_high_watermark,
-                deltacat_storage,
-            )
-        if new_partition:
-            partition = new_partition
-            destination_partition_locator = new_partition.locator
-            compaction_rounds_executed += 1
-        # Take new primary key index sizes into account for subsequent compaction rounds and their dedupe steps
-        if new_rci:
-            min_pk_index_pa_bytes = new_rci.pk_index_pyarrow_write_result.pyarrow_bytes
-
-    logger.info(f"Partition-{source_partition_locator.partition_values}-> Compaction session data processing completed in "
-                f"{compaction_rounds_executed} rounds.")
+    new_partition, new_rci, new_rcf_s3_url = \
+        _execute_compaction_round(
+            source_partition_locator,
+            destination_partition_locator,
+            primary_keys,
+            compaction_artifact_s3_bucket,
+            last_stream_position_to_compact,
+            hash_bucket_count,
+            sort_keys,
+            records_per_compacted_file,
+            input_deltas_stats,
+            min_hash_bucket_chunk_size,
+            compacted_file_content_type,
+            pg_config,
+            schema_on_read,
+            rebase_source_partition_locator,
+            rebase_source_partition_high_watermark,
+            deltacat_storage,
+        )
+    if new_partition:
+        partition = new_partition
```

Comment:
Minor: `partition` variable seems redundant now.

Reply:
Doesn't hurt to just keep it, in future we may merge back the multi-rounds.

Reply:
Yes, it was minor comment but I believe it's a very small change that improves readability.